### PR TITLE
Adds docs for cloud target discovery

### PR DESCRIPTION
**Pull request by geofflamrock**
This PR makes a start on adding documentation for the new dynamic infrastructure being worked in #project-dynamic-infrastructure. 

~~At the moment it's put into a section called "Reflected targets" although the name for the feature hasn't been finalised. Internally we are calling it Dynamic Infrastructure however there are already docs for the existing feature with the same name and we don't really want to confuse users.~~ edit: We've changed to using "Cloud target discovery" as a name, at least for the time being.

### CODE REVIEW

File: docs/infrastructure/deployment-targets/reflected-targets/index.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+- `octopus-environment`: The name of the [environment](/docs/infrastructure/environments) the target can be used in during deployments. This tag is required and only deployments for the environment will discover the target.
+- `octopus-role`: The [role](/docs/infrastructure/deployment-targets#target-roles) that should be applied to the target. This tag is required and only deployments with a step that matches the role will discover the target.
+- `octopus-space`: The name of the [space](/docs/administration/spaces) the target can be used in. This tag is optional and if present only deployments within the matching space can discover the target.
+- `octopus-project`: The name of the [project](/docs/projects) the target will be discovered by. This tag is optional and if present only deployments for the matching project can discover the target.
+- `octopus-tenant`: The name of the [tenant](/docs/projects) the target can be discovered for. This tag is optional and if present only deployments for the matching tenant will discover the target.
+
+## Examples
+
+### ECS Cluster
```
Comment (geofflamrock) [inline docs/infrastructure/deployment-targets/reflected-targets/index.md:60]: I added an ECS cluster example here instead of an Azure one as feels like we don't want to appear to be too Azurey/Microsofty. The first piece of work is for reflected Azure Web App targets so I suppose we could add that first and then add an ECS example afterwards when that is available.

File: docs/infrastructure/deployment-targets/reflected-targets/index.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
+---
+title: Reflected Targets
+description: Octopus resources created in cloud providers can be discovered automatically by Octopus
+position: 100
+---
+
+Octopus can automatically discover deployment targets during deployment using tags added to cloud resources. These targets that are automatically discovered will also be removed from Octopus when the associated cloud resource is removed. Automatic discovery will take place before the relevant deployment step and can be used as part of deployment processes that create your cloud infrastructure before deployment.
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/reflected-targets/index.md:7]: I feel we can drop the word `automatically` in these sentences - it does not seem to add more clarity / context.

Comment (andrewabest) [reply docs/infrastructure/deployment-targets/reflected-targets/index.md:7]: > relevant deployment step

This may need a small amount of clarification. Perhaps it could just be simplified to "before deployment steps"

Comment (andrewabest) [reply docs/infrastructure/deployment-targets/reflected-targets/index.md:7]:
```suggestion
Octopus can discover deployment targets during deployments using tags added to your cloud resources. Octopus will remove any discovered targets when the associated cloud resource is removed. Target discovery takes place before deployment steps, and is useful when your deployment process creates your target cloud infrastructure before deploying software to it.
```

Comment (andrewabest) [reply docs/infrastructure/deployment-targets/reflected-targets/index.md:7]: Serving suggestion

File: docs/infrastructure/deployment-targets/reflected-targets/index.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+:::hint
+Using automatic discovery of deployment targets is recommended over the existing [dynamic infrastructure](/docs/infrastructure/deployment-targets/dynamic-infrastructure) functionality.
+:::
+
+To enable automatic discovery and clean up of targets use the following steps:
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/reflected-targets/index.md:13]: Suggestion: I'd love for these documents to read like our goal:

> Just configure your credentials, and ensure your cloud resources are tagged

That would obviously still need the detail, but I'd like for that to be the first things users read. 

Then after that we could add a "For existing projects" section and talk about the need to enable dynamic infrastructure and set deployments without targets settings if not already done.

This would help new users not having to chew over our legacy to get to what they wanted.

File: docs/infrastructure/deployment-targets/reflected-targets/index.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+- `octopus-tenant`: The name of the [tenant](/docs/projects) the target can be discovered for. This tag is optional and if present only deployments for the matching tenant will discover the target.
+
+## Examples
+
+### ECS Cluster
+
+As an example, let's say you have an project in Octopus called _Pet Shop_ that deploys an application to an ECS cluster in a _Development_ environment using a role of _PetShopFrontEnd_ and this cluster is dynamically created as part of the deployment using a CloudFormation template.
+
+To use this ECS cluster previously in Octopus you might have had to put in a [manual intervention step](/docs/projects/built-in-step-templates/manual-intervention-and-approvals) and manually create the target, or add a [script step](/docs/infrastructure/deployment-targets/dynamic-infrastructure/new-octopustarget.md) with custom code to try and find and create the ECS cluster target. In addition, previously when this cluster was no longer needed you might have needed to either [run a script](/docs/infrastructure/deployment-targets/dynamic-infrastructure/remove-octopustarget.md) or manually remove the target in Octopus.
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/reflected-targets/index.md:64]: Not sure if anyone ever added a manual intervention to go in and registered targets - but they would certinly have had to have manually registered the target - perhaps we can simplify the first part to just that?

File: docs/infrastructure/deployment-targets/reflected-targets/index.md (hunk 3 above)
Comment (geofflamrock) [reply docs/infrastructure/deployment-targets/reflected-targets/index.md:13]: Yeah nice I like it 👍

File: docs/infrastructure/deployment-targets/amazon-ecs-cluster-target.md [hunk 5]
```diff
@@ -6,10 +6,24 @@ position: 30
 
 ECS Cluster targets are used by the [ECS steps](/docs/deployments/aws/index.md) to define the context in which deployments and scripts are run.
 
+:::hint
```
Comment (geofflamrock) [inline docs/infrastructure/deployment-targets/amazon-ecs-cluster-target.md:9]: I'm thinking might actually need to split this out into a separate PR as we are planning on shipping changes for Azure target discovery before ECS.

File: docs/infrastructure/deployment-targets/azure/web-app-targets/index.md [hunk 6]
```diff
@@ -6,12 +6,26 @@ position: 20
...
 
 - You need an [Azure Service Principal account](/docs/infrastructure/accounts/azure/index.md#azure-service-principal) that references your Azure subscription.
 
 - Once your Azure account is setup, you need an existing Azure Web App / App Service setup within your Azure subscription. To learn more about App Services, see the [Azure App Services documentation](https://docs.microsoft.com/en-us/azure/app-service/) that can help you get started. If you are dynamically creating the web app during your deployment, check our section about [creating Web App targets by scripts using service messages](#creating-web-app-targets-by-scripts).
 
+## Discovering web app targets
+
+Octopus can discover Azure Web App targets as part of your deployment using tags on your resource. To discover targets use the following steps.
+
+- Add an Azure account variable **Octopus.Azure.Account** to your project.
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/azure/web-app-targets/index.md:23]:
```suggestion
- Add an Azure account variable named **Octopus.Azure.Account** to your project.
```

File: docs/infrastructure/deployment-targets/azure/web-app-targets/index.md [hunk 7]
```diff
@@ -6,12 +6,26 @@ position: 20
...
+- Add tags to your Azure Web App so that Octopus can match it to your deployment step and environment.
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/azure/web-app-targets/index.md:24]: I feel like we should link the "Add tags" words here to the relevant doco, otherwise it feels like an immediate questionmark

File: docs/infrastructure/deployment-targets/azure/web-app-targets/index.md [hunk 8]
```diff
@@ -6,12 +6,26 @@ position: 20
...
+- Add a "Deploy an Azure App Service" or "Deploy an Azure Web App (Web Deploy)" step to your deployment process. The target role on the step will be used along with the environment being deployed to, to discover Azure Web App targets and then deploy to them.
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/azure/web-app-targets/index.md:25]:
```suggestion
- Add a "Deploy an Azure App Service" or "Deploy an Azure Web App (Web Deploy)" step to your deployment process. During deployment, the target role on the step will be used along with the environment being deployed to, to discover Azure Web App targets to deploy to.
```

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
+---
+title: Cloud Target Discovery
+description: Resources created in cloud providers can be discovered as deployment targets by Octopus
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:3]:
```suggestion
description: Cloud resources can be discovered and registered as deployment targets by Octopus
```

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+position: 100
+---
+
+:::hint
+The Early Access Preview of the cloud target discovery feature was added in **Octopus 2022.1** and is enabled via **{{Configuration, Features}}**.
+:::
+
+Octopus can discover deployment targets during deployments using tags added to your cloud resources. Octopus will remove any discovered targets when the associated cloud resource is removed. Target discovery takes place before deployment steps, and is useful when your deployment process creates your target cloud infrastructure before deploying software to it.
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:11]:
```suggestion
Octopus can discover deployment targets during deployments using tags added to your cloud resources. Octopus will remove any discovered targets when the associated cloud resource is removed. Target discovery takes place during deployment, and is useful when your deployment process creates your target cloud infrastructure before deploying software to it.
```

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+:::hint
+Using discovery of deployment targets is recommended over the existing [dynamic infrastructure](/docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md) functionality.
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:14]:
```suggestion
We recommend cloud target discovery over the existing [dynamic infrastructure](/docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md) functionality.
```

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+| `octopus-space`       | No       | The name of the [space](/docs/administration/spaces/index.md) the target can be used in. If present only deployments within the matching space can discover the target.                      | `octopus-space = PetShopTeam`       |
+| `octopus-project`     | No       | The name of the [project](/docs/projects/index.md) the target will be discovered by. If present only deployments for the matching project can discover the target.                           | `octopus-project = PetShop`         |
+| `octopus-tenant`      | No       | The name of the [tenant](/docs/projects/index.md) the target can be discovered for. If present only deployments for the matching tenant will discover the target.                            | `octopus-tenant = MyClient`         |
+
+## Add step to deployment process
+
+Octopus will discover targets if one of the following steps are in your deployment process. Each step will discover targets that match it's target role before the step is run.
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:45]:
```suggestion
Octopus will discover targets if one of the following steps are in your deployment process. Each step will discover targets that match it's target role.
```

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+
+- Deploy an Azure App Service
+- Deploy an Azure Web App (Web Deploy)
+
+## Enabling discovery for existing projects
+
+Target discovery is enabled for all new projects by default, to enable it for existing projects use the following steps.
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:52]:
```suggestion
Target discovery is enabled for all new projects by default. 

If you have an existing project, enable it with the following steps.
```

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+
+### Enable dynamic infrastructure
+
+To discover targets for an environment, dynamic infrastructure needs to be enabled. To enable for an existing environment:
+
+1. Navigate to **{{Infrastructure,Environments}}**.
+1. Click the ... overflow menu for the environment you want to enable or disable dynamic infrastructure on and select **Edit**.
+1. Expand the **Dynamic infrastructure** section and tick or untick the check-box to enable or disable managing dynamic infrastructure.
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:60]:
```suggestion
1. Expand the **Dynamic infrastructure** section and tick the check-box to enable dynamic infrastructure.
```

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+1. Click **SAVE**.
+
+### Enable deployments without a target
+
+Using target discovery during a deployment means that there may be no existing targets at the start of a deployment. To allow deployments to start without any targets for an existing project:
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:65]:
```suggestion
Using target discovery during a deployment means that there may be no existing targets at the start of a deployment. To allow deployments to start without any targets:
```

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+
+## Examples
+
+### Azure Web App
+
+Let's say you have an project in Octopus called _Pet Shop_ that deploys an application to an Azure Web App in a _Development_ environment using a role of _PetShopFrontEnd_ and this web app is dynamically created as part of the deployment using an ARM template.
+
+To use this web app previously in Octopus you might have either registered the target manually, or used a [script step](/docs/infrastructure/deployment-targets/dynamic-infrastructure/azure-web-app-target.md) with custom code to try and find and create the web app target. In addition, previously when this web app was no longer needed you might have needed to either [run a script](/docs/infrastructure/deployment-targets/dynamic-infrastructure/remove-octopustarget.md) or manually remove the target in Octopus.
+
+By configuring credentials in a variable and adding tags to the deployment of the web app Octopus can discover this target for you and also remove the target from Octopus when it is removed in Azure.
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:79]:
```suggestion
By configuring a well-known variable and tagging your Azure Web App appropriately, Octopus can discover this target for you at deployment time. Additionally, Octopus will continue to monitor the target, and will remove it if it is removed in Azure.
```

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+
+- Configure an [Azure account](/docs/projects/variables/azure-account-variables.md) variable in your project named **Octopus.Azure.Account**, selecting an account that has permissions to be able to find the web app.
+- Add tags to the ARM template for the web app to allow Octopus to discover it. For our example we can add the following tags to ensure that it is discovered correctly by our (and only by our project) using [variable substitution](/docs/projects/variables/variable-substitutions.md):
```
Comment (andrewabest) [inline docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:82]:
```suggestion
- Add tags to the web app resource within the ARM template to allow Octopus to discover it. For our example we can add the following tags to ensure that it is discovered correctly by our (and only by our project) using [variable substitution](/docs/projects/variables/variable-substitutions.md):
```

**Review by andrewabest — APPROVED** [review]
Looks good @geofflamrock!

File: docs/infrastructure/deployment-targets/azure/web-app-targets/index.md [hunk 18]
```diff
@@ -6,12 +6,26 @@ position: 20
...
 
+## Discovering web app targets
+
+Octopus can discover Azure Web App targets as part of your deployment using tags on your resource. To discover targets use the following steps.
+
+- Add an Azure account variable named **Octopus.Azure.Account** to your project.
+- [Add tags](/docs/infrastructure/deployment-targets/cloud-target-discovery/index.md#tag-cloud-resources) to your Azure Web App so that Octopus can match it to your deployment step and environment.
+- Add a "Deploy an Azure App Service" or "Deploy an Azure Web App (Web Deploy)" step to your deployment process. During deployment, the target role on the step will be used along with the environment being deployed to, to discover Azure Web App targets to deploy to.
```
Comment (harrisonmeister) [inline docs/infrastructure/deployment-targets/azure/web-app-targets/index.md:25]:
```suggestion
- Add a `Deploy an Azure App Service` or `Deploy an Azure Web App (Web Deploy)` step to your deployment process. During deployment, the target role on the step will be used along with the environment being deployed to, to discover Azure Web App targets to deploy to.
```

File: docs/infrastructure/deployment-targets/azure/web-app-targets/index.md [hunk 19]
```diff
@@ -6,12 +6,26 @@ position: 20
...
+
 ## Requirements
 
 - You need an [Azure Service Principal account](/docs/infrastructure/accounts/azure/index.md#azure-service-principal) that references your Azure subscription.
 
 - Once your Azure account is setup, you need an existing Azure Web App / App Service setup within your Azure subscription. To learn more about App Services, see the [Azure App Services documentation](https://docs.microsoft.com/en-us/azure/app-service/) that can help you get started. If you are dynamically creating the web app during your deployment, check our section about [creating Web App targets by scripts using service messages](#creating-web-app-targets-by-scripts).
 
+## Discovering web app targets
+
+Octopus can discover Azure Web App targets as part of your deployment using tags on your resource. To discover targets use the following steps.
```
Comment (harrisonmeister) [inline docs/infrastructure/deployment-targets/azure/web-app-targets/index.md:21]:
```suggestion
Octopus can discover Azure Web App targets as part of your deployment using tags on your resource. To discover targets use the following steps:
```

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md [hunk 20]
```diff
@@ -0,0 +1,98 @@
...
+The Early Access Preview of the cloud target discovery feature was added in **Octopus 2022.1** and is enabled via **{{Configuration, Features}}**. The preview does not currently include removal of targets when the cloud resource is removed, this is on our roadmap.
+:::
+
+Octopus can discover deployment targets during deployments using tags added to your cloud resources. Target discovery takes place during deployment, and is useful when your deployment process creates your target cloud infrastructure before deploying software to it.
+
+:::hint
+We recommend cloud target discovery over the existing [dynamic infrastructure](/docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md) functionality.
+:::
+
+To enable discovery use the following steps.
```
Comment (harrisonmeister) [inline docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:17]:
```suggestion
This page will walk you through the steps needed to enable cloud target discovery.
```

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md [hunk 21]
```diff
@@ -0,0 +1,98 @@
...
+Target discovery is enabled for all new projects and environments by default.
+
+If you have an existing project or environment, enable it with the following steps.
+
+### Enable dynamic infrastructure
+
+To discover targets for an environment, dynamic infrastructure needs to be enabled. To enable for an existing environment:
+
+1. Navigate to **{{Infrastructure,Environments}}**.
+1. Click the ... overflow menu for the environment you want to enable or disable dynamic infrastructure on and select **Edit**.
```
Comment (harrisonmeister) [inline docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:61]:
```suggestion
1. Click the overflow menu (`...`) for the environment you want to enable or disable dynamic infrastructure on and select **Edit**.
```

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md [hunk 22]
```diff
@@ -0,0 +1,98 @@
...
+1. Expand the **Dynamic infrastructure** section and tick the check-box to enable dynamic infrastructure.
+1. Click **SAVE**.
+
+### Enable deployments without a target
+
+Using target discovery during a deployment means that there may be no existing targets at the start of a deployment. To allow deployments to start without any targets:
+
+1. Navigate to **{{Projects,{Project name},Deployments,Settings}}**.
+1. Expand the **Deployment Targets Required** and select the "Allow deployments to be created when there are no deployment targets" option.
```
Comment (harrisonmeister) [inline docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:70]:
```suggestion
1. Expand the **Deployment Targets Required** and select the **Allow deployments to be created when there are no deployment targets** option.
```

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md [hunk 23]
```diff
@@ -0,0 +1,98 @@
...
+    //...otherProperties,
+    "tags": {
+        "octopus-environment": "#{Octopus.Environment.Name}",
+        "octopus-role": "PetShopFrontEnd",
+        "octopus-project": "#{Octopus.Project.Name}"
+    }
+}]
+```
+
+Octopus will now discover the web app as a target before deploying to it, matching the environment, role and project from the deployment to the tags created with the ARM template, without any custom scripts or manual registration! Octopus will also remove this target if it is later removed from Azure.
```
Comment (harrisonmeister) [inline docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:98]: This might not be an issue, but the last sentence contradicts the hint at the start of the page referring to functionality on removing targets not being available in the EAP. If the functionality will be available before the doco goes live, then no issue. If not, we might want to highlight this again, or change the wording?

```suggestion
Octopus will now discover the web app as a target before deploying to it, matching the environment, role, and project from the deployment to the tags created with the ARM template, without any custom scripts or manual registration! Octopus will also remove this target if it is later removed from Azure in future releases of this feature.
```

File: docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md [hunk 24]
```diff
@@ -4,6 +4,10 @@ description: Octopus resources created in cloud providers can also be modeled in
 position: 100
 ---
 
+:::hint
+Octopus version 2022.1 has support for discovering and cleaning up supported types of deployment targets using tags on cloud resources, please see our documentation on [Cloud Target Discovery](/docs/infrastructure/deployment-targets/cloud-target-discovery/index.md) for more information.
```
Comment (harrisonmeister) [inline docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md:8]: Should this hint also expand on our recommendation made [here](https://github.com/OctopusDeploy/docs/pull/1448/files#diff-3dbf529eab1cf9b9fe14638f80e58b3eb35b1bb9729aaa734d1c7b360173db87R13) e.g.

> We recommend cloud target discovery over the existing [dynamic infrastructure](/docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md) functionality.

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md [hunk 25]
```diff
@@ -0,0 +1,98 @@
...
+
+## Configure credentials for discovery
+
+To perform discovery of targets from cloud resources, Octopus uses a well-known set of variables configured within your project to provide the credentials for authentication to the cloud provider. Variables can be [scoped](/docs/projects/variables/index.md#scoping-variables) allowing for different credentials to be used when performing discovery as required (for example in different environments).
+
+The variables used are different for each supported cloud provider.
+
+### Azure
+
+To discover Azure cloud resources, Octopus uses an [Azure account](/docs/projects/variables/azure-account-variables.md) variable named **Octopus.Azure.Account**.
```
Comment (harrisonmeister) [inline docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:27]: I wonder if this section either needs to labour the point of the name in stronger wording, or a hint to say something like:

> If you have another Azure account variable named differently, Octopus won't look for Azure cloud resources to discover. The account must be named **Octopus.Azure.Account**.

My reasoning is I can imagine a scenario where a customer doesn't see the importance of the name of the variable, and instead just *assumes* it would work if they have any Azure account variable.

**Review by harrisonmeister — APPROVED** [review]
Looks good, I've made some comments, but none are deal breakers :)

**Comment by geofflamrock** [comment]
> > As the project is not user-facing yet we don't want the documentation to be released, so please do not merge this.
> 
> @geofflamrock - I've applied the **pending-octopus-release** label to this :)

Thanks @harrisonmeister , we are actually ready to merge these in now, so I'll remove the label and apply those changes you suggested :)

File: docs/infrastructure/deployment-targets/cloud-target-discovery/index.md (hunk 25 above)
Comment (geofflamrock) [reply docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:27]: Yep it's prob good to be extra specific here, I might change it to:

```suggestion
To discover Azure cloud resources, Octopus will look for a specifically named [Azure account](/docs/projects/variables/azure-account-variables.md) variable **Octopus.Azure.Account**.
```

Comment (harrisonmeister) [reply docs/infrastructure/deployment-targets/cloud-target-discovery/index.md:27]: Yes, I like that much more than my hint 😄

File: docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md (hunk 24 above)
Comment (geofflamrock) [reply docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md:8]: Hmmm good thought, although given the feature is EAP at the moment I wonder if we should make strong recommendations on it's use over existing features at all? Do you have any thoughts on this @andrewabest ?

Comment (harrisonmeister) [reply docs/infrastructure/deployment-targets/dynamic-infrastructure/index.md:8]: This wasn't a big deal for me either way. So happy with whatever you decide 😃
